vpr/src/route/rr_graph_clock.h

```diff
@@ -0,0 +1,119 @@
+#ifndef RR_GRAPH_CLOCK_H
+#define RR_GRAPH_CLOCK_H
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <unordered_map>
+#include <set>
+#include <utility>
+
+#include "clock_fwd.h"
+
+#include "clock_network_types.h"
+#include "clock_connection_types.h"
+
+class ClockNetwork;
+class ClockConnection;
+
+class SwitchPoint {
+    public:
+        // [grid_width][grid_height][nodes]
+        std::vector<std::vector<std::vector<int>>> rr_node_indices;
+        std::set<std::pair<int, int>> locations; // x,y
+    public:
+        /** Getters **/
+        std::vector<int> get_rr_node_indices_at_location(int x, int y) const;
+
+        std::set<std::pair<int, int>> get_switch_locations() const;
+
+        /** Setters **/
+        void insert_node_idx(int x, int y, int node_idx);
+};
+
+class SwitchPoints {
+    public:
+        std::unordered_map<std::string, SwitchPoint> switch_name_to_switch_location;
+    public:
+        /** Getters **/
+        std::vector<int> get_rr_node_indices_at_location(
+            std::string switch_name,
+            int x,
+            int y) const;
+
+        std::set<std::pair<int, int>> get_switch_locations(std::string switch_name) const;
+
+        /** Setters **/
+        void insert_switch_node_idx(std::string switch_name, int x, int y, int node_idx);
+};
+
+class ClockRRGraph {
+
+    public:
+        /* Returns the current ptc num where the wire should be drawn and updates the
+           channel width. */
+        int get_and_increment_chanx_ptc_num();
+        int get_and_increment_chany_ptc_num();
```

Comment:
Add comment saying how this works and that it updates some global state.
Make load_rr_graph update the chanwidth_x etc. data structures so drawing works.

Reply:
Done: commit da830a1c6d4b10cd2dc1b53c17cb94517458dd46